src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,52 @@
+import React, { ChangeEvent } from 'react';
+import { SearchLink } from './SearchLink';
+import cn from 'classnames';
+import { useSearchParams } from 'react-router-dom';
+import { getSearchWith } from '../utils/searchHelper';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || '';
+  const searchQuery = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const CENTURIES = ['16', '17', '18', '19', '20'];
+
+  function handleInputChange(event: ChangeEvent<HTMLInputElement>) {
+    const query = event.target.value.trimStart() || null;
+    setSearchParams(getSearchWith(searchParams, { query }));
+  }
+
+  const handleCenturyParams = (selectedCentury: string) =>
+    centuries.includes(selectedCentury)
+      ? centuries.filter(century => century !== selectedCentury)
+      : [...centuries, selectedCentury];
```

Comment:
The `handleCenturyParams` function is intended to toggle a century in the `centuries` array. However, it currently only returns the new array but does not update the search parameters. You should call `setSearchParams` with the updated centuries array to ensure the URL reflects the current filter state.